frontend/awx/resources/templates/TemplatePage/TemplateLaunchWizard.tsx

```diff
@@ -0,0 +1,323 @@
+          organizationId={template.organization?.toString() ?? ''}
+        />
+      ),
+      hidden: () => !template.ask_execution_environment_on_launch,
+    },
+    {
+      label: 'Instance Groups',
+      id: 'instance-groups',
+      hidden: () => !template.ask_instance_groups_on_launch,
+      inputs: (
+        <PageFormInstanceGroupSelect<TemplateLaunch>
+          name="instance_groups"
+          labelHelp={t(`Select the instance groups for this template to run on.`)}
+        />
+      ),
+    },
+    {
+      label: 'Other prompts',
+      id: 'other-prompts',
+      hidden: () => !shouldShowOtherStep(config),
+      inputs: <OtherPromptsStep config={config} template={template} />,
+    },
+    {
+      label: 'Review',
+      id: 'review',
+      element: <TemplateLaunchReviewStep template={template} />,
+    },
+  ];
+
+  const { defaults } = config;
+
+  const readOnlyLabels = defaults?.labels?.map((label) => ({
+    ...label,
+    isReadOnly: true,
+  }));
+
+  const initialValues = {
+    inventory: {
+      inventory: defaults.inventory.id ? defaults.inventory : null,
+    },
+    credentials: {
+      credentials: defaults.credentials,
+    },
+    'credential-passwords': {},
+    'execution-environment': {
+      execution_environment: defaults.execution_environment,
+    },
+    'instance-groups': {
+      instance_groups: defaults.instance_groups,
+    },
+    'other-prompts': {
+      diff_mode: defaults.diff_mode,
+      scm_branch: defaults.scm_branch,
+      extra_vars: defaults.extra_vars,
+      forks: defaults.forks,
+      job_slice_count: defaults.job_slice_count,
+      job_tags: parseStringToTagArray(defaults.job_tags),
+      job_type: defaults.job_type,
+      labels: readOnlyLabels,
+      limit: defaults.limit,
+      skip_tags: parseStringToTagArray(defaults.skip_tags),
+      timeout: defaults.timeout,
+      verbosity: defaults.verbosity,
+    },
+  };
+
+  return (
+    <PageLayout>
+      <PageHeader
+        title={t('Prompt on Launch')}
+        breadcrumbs={[
+          { label: t('Templates'), to: getPageUrl(AwxRoute.Templates) },
+          { label: t('Templates') },
+        ]}
+      />
+      <PageWizard<TemplateLaunch>
+        steps={steps}
+        defaultValue={initialValues}
+        onSubmit={handleSubmit}
+      />
+    </PageLayout>
+  );
+}
+
+function shouldShowOtherStep(launchData: LaunchConfiguration) {
+  return (
+    launchData.ask_job_type_on_launch ||
+    launchData.ask_limit_on_launch ||
+    launchData.ask_verbosity_on_launch ||
+    launchData.ask_tags_on_launch ||
+    launchData.ask_skip_tags_on_launch ||
+    launchData.ask_variables_on_launch ||
+    launchData.ask_scm_branch_on_launch ||
+    launchData.ask_diff_mode_on_launch ||
+    launchData.ask_labels_on_launch ||
+    launchData.ask_forks_on_launch ||
+    launchData.ask_job_slice_count_on_launch ||
+    launchData.ask_timeout_on_launch
+  );
+}
+
+export function parseStringToTagArray(str: string) {
```

Comment:
This function is already defined [here](https://github.com/ansible/ansible-ui/blob/main/frontend/awx/resources/templates/JobTemplateFormHelpers.ts#L75). Could we reuse that?

Reply:
Pushed an update that removes duplicate fn `parseStringToTagArray` definitions.